### PR TITLE
[audio_dac] [aic3204] Add new component + platform docs

### PR DESCRIPTION
**Pull request by kbx81**
## Description:

SSIA

**Related issue (if applicable):** fixes <link to issue>

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#7505

## Checklist:

  - [x] I am merging into `next` because this is new documentation that has a matching pull-request in [esphome](https://github.com/esphome/esphome) as linked above.  
    or
  - [ ] I am merging into `current` because this is a fix, change and/or adjustment in the current documentation and is not for a new component or feature.

  - [x] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

**Comment by kbx81** [comment]
@esphomebot generate image AIC3204

**Comment by kbx81** [comment]
@esphomebot generate image Audio DAC

**Comment by kbx81** [comment]
@esphomebot generate image "Audio DAC"

**Comment by kbx81** [comment]
@esphomebot generate image Audio\ DAC

**Comment by kbx81** [comment]
@esphomebot generate image 'Audio DAC'

**Comment by kbx81** [comment]
@esphomebot generate image `Audio DAC`

File: components/audio_dac/index.rst [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+See Also
+--------
+
+.. toctree::
+    :maxdepth: 1
+    :glob:
+
+    *
+
+- :ghedit:`Edit`
```
Comment (jesserockz) [inline components/audio_dac/index.rst]:
```suggestion
Platforms
---------

.. toctree::
    :maxdepth: 1
    :glob:

    *

See Also
--------

- :ghedit:`Edit`
```

**Review by jesserockz — APPROVED** [review]

**Comment by nagyrobi** [comment]
Wow! This chip supports multiple analog audio inputs too, and has a built-in audio mixer! (according to [spec sheet](https://www.ti.com/lit/ds/symlink/tlv320aic3204.pdf))

Could be a good all-in-one solution for https://github.com/esphome/feature-requests/issues/1750 and https://github.com/esphome/feature-requests/issues/1751 (use case described [here](https://forum.raspiaudio.com/t/suggestion-for-espmuse-multiple-analog-inputs/401/7))

**Comment by jesserockz** [comment]
@nagyrobi https://github.com/esphome/home-assistant-voice-pe/blob/dev/home-assistant-voice.yaml#L1338-L1339

**Pull request merged**